presto-orc/src/main/java/com/facebook/presto/orc/metadata/OrcType.java

```diff
@@ -295,4 +297,29 @@ public static List<OrcType> createOrcRowType(int nextFieldTypeIndex, List<String
 
         return orcTypes;
     }
+
+    /**
+     * Converts zero-based column numbers to ORC node indexes.
+     * NOTE: This method ignores columns that are not included into the ORC types.
+     */
+    public static Set<Integer> mapColumnToNode(Set<Integer> columnIndexes, List<OrcType> orcTypes)
+    {
+        requireNonNull(columnIndexes, "columnIndexes is null");
+        requireNonNull(orcTypes, "orcTypes is null");
+
+        if (columnIndexes.isEmpty()) {
+            return ImmutableSet.of();
+        }
+
+        ImmutableSet.Builder<Integer> nodes = ImmutableSet.builder();
+        OrcType rootType = orcTypes.get(0);
+
+        for (int columnIndex = 0; columnIndex < rootType.getFieldCount(); columnIndex++) {
```

Comment:
This need not be addressed. 

But instead of iterating over the columnId, we can directly iterate the columnIndex, but this one supports columnIndex even if they are out of bonds from fieldCount. So this can be this way.